share/functions/cd.fish

```diff
@@ -3,10 +3,20 @@
 #
 function cd --description "Change directory"
     set -l MAX_DIR_HIST 25
+    set -l options h/help L P
+    argparse -n cd --max-args=3 $options -- $argv
+    or return
 
-    if test (count $argv) -gt (test "$argv[1]" = "--" && echo 2 || echo 1)
-        printf "%s\n" (_ "Too many args for cd command")
-        return 1
+    # Drop the -L argument as fish always resolves the directory. Like POSIX
+    # it's the default option for cd.
+    if set -q _flag_L
+        set argv (string replace -- '-L' '' $argv)
+    end
+
+    # Drop the argument and use the realpath
+    if set -q _flag_P
+        set argv (string replace -- '-P' '' $argv)
```

Comment:
This line can be dropped, see above.

Reply:
If I have not this line I have en error:

    cd: Unknown option “-P”

    /usr/share/fish/functions/cd.fish (line 35):
        builtin cd $argv
        ^
    in function 'cd' with arguments '-P ansiconf/'

So I think we need this line.

Reply:
No sorry I said something wrong.